src/jit/codegenlegacy.cpp

```diff
@@ -18849,29 +18849,48 @@ regMaskTP CodeGen::genCodeForCall(GenTreeCall* call, bool valUsed)
                 // stub dispatching is off or this is not a virtual call (could be a tailcall)
                 {
                     regNumber vptrReg;
+                    regNumber vptrReg1;
+                    regMaskTP vptrMask1;
                     unsigned  vtabOffsOfIndirection;
                     unsigned  vtabOffsAfterIndirection;
+                    bool      isRelative;
 
                     noway_assert(callType == CT_USER_FUNC);
 
+                    /* Get hold of the vtable offset (note: this might be expensive) */
+
+                    compiler->info.compCompHnd->getMethodVTableOffset(call->gtCallMethHnd, &vtabOffsOfIndirection,
+                                                                      &vtabOffsAfterIndirection, &isRelative);
+
                     vptrReg =
                         regSet.rsGrabReg(RBM_ALLINT); // Grab an available register to use for the CALL indirection
                     vptrMask = genRegMask(vptrReg);
 
+                    if (isRelative)
+                    {
+                        vptrReg1  = regSet.rsGrabReg(RBM_ALLINT & ~vptrMask);
+                        vptrMask1 = genRegMask(vptrReg1);
+                    }
+
```

Comment:
Normally, if you are going to grab a register, you want to make sure the register allocator has tried to make one available, to avoid causing spilling. This would require changing regalloc.cpp, around line 4911 in rpPredictTreeRegUse(). However, the call above to `getMethodVTableOffset` has the comment "this might be expensive". So maybe we don't want to do it twice. However, if there are no registers available here, since this wasn't done, it might require spilling a register.